storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,265 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
-        } else {
-          executorInfoBuilder
-              .setCommand(CommandInfo.newBuilder()
-                  .addUris(URI.newBuilder().setValue((String) _conf.get(CONF_EXECUTOR_URI)))
-                  .addUris(URI.newBuilder().setValue(configUri))
-                  .setValue(commandLineShim.getCommandLine(details.getId())));
-        }
+        try {
+          List<ResourceEntry> scalarResourceEntryList = null;
+          List<ResourceEntry> rangeResourceEntryList = null;
+
+          if (subtractExecutorResources) {
+            scalarResourceEntryList = offerResources.reserveAndGet(ResourceType.CPU, new ScalarResourceEntry(executorCpu));
+            executorCpuResources.addAll(createMesosScalarResource(ResourceType.CPU, scalarResourceEntryList));
+            scalarResourceEntryList = offerResources.reserveAndGet(ResourceType.MEM, new ScalarResourceEntry(executorMem));
+            executorMemResources.addAll(createMesosScalarResource(ResourceType.MEM, scalarResourceEntryList));
+          } else {
+            // TODO(ksoundararaj): Do we need to ensure consistent roles for all executors belonging to same topology?
+            // If so, how do we do that?
+            executorCpuResources.add(createMesosScalarResource(ResourceType.CPU, new ScalarResourceEntry(executorCpu)));
+            executorMemResources.add(createMesosScalarResource(ResourceType.MEM, new ScalarResourceEntry(executorMem)));
+          }
 
-        LOG.info("Launching task with Mesos Executor data: < {} >", executorDataStr);
-        TaskInfo task = TaskInfo.newBuilder()
-            .setName(taskName)
-            .setTaskId(taskId)
-            .setSlaveId(offer.getSlaveId())
-            .setExecutor(executorInfoBuilder.build())
-            .addAllResources(workerCpuResources)
-            .addAllResources(workerMemResources)
-            .addAllResources(workerPortsResources)
-            .build();
+          /**
+           *  Autostart logviewer cannot be supported. If we try to fetch a random port P
+           *  from the list of available ports and assign it to the logviewer, its possible that
+           *  one of the workers is already assigned to that port. So I am not supporting auto starting
+           *  logviewer at this point
+           *
+          if (MesosCommon.autoStartLogViewer(mesosStormConf)) {
+            Long logViewerPort = offerResources.reserveRangeResource(ResourceType.PORTS, 1).get(0);
+            List<RangeResourceEntry> logViewerPortList = new ArrayList<>();
+            logViewerPortList.add(new RangeResourceEntry(logViewerPort, logViewerPort));
+            executorPortResources.add(createMesosRangeResource(ResourceType.PORTS, logViewerPortList));
+          } */
+
+          String supervisorStormLocalDir = (String) mesosStormConf.get(CONF_MESOS_SUPERVISOR_STORM_LOCAL_DIR);
+          if (supervisorStormLocalDir != null) {
+            extraConfig += String.format(" -c storm.local.dir=%s", supervisorStormLocalDir);
```

Comment:
hah. yeah thats totally wrong

```
   // Set the MesosSupervisor's storm.local.dir value. By default it is set to "storm-local",
  // which is the same as the storm-core default, which is interpreted relative to the pwd of
  // the storm daemon (the mesos executor sandbox in the case of the MesosSupervisor).
  //
  // If this isn't done, then MesosSupervisor inherits MesosNimbus's storm.local.dir config.
  // That is fine if the MesosNimbus storm.local.dir config is just the default too ("storm-local"),
  // since the mesos executor sandbox is isolated per-topology.
  // However, if the MesosNimbus storm.local.dir is specialized (e.g., /var/run/storm-local), then
  // this will cause problems since multiple topologies on the same host would use the same
  // storm.local.dir and thus interfere with each other's state. And *that* leads to one topology's
  // supervisor killing the workers of every other topology on the same host (due to interference
  // in the stored worker assignments).
  // Note that you *can* force the MesosSupervisor to use a specific directory by setting
  // the "mesos.supervisor.storm.local.dir" variable in the MesosNimbus's storm.yaml.
  public String getStormLocalDirForWorkers() {
    String supervisorStormLocalDir = (String) mesosStormConf.get(CONF_MESOS_SUPERVISOR_STORM_LOCAL_DIR);
    if (supervisorStormLocalDir == null) {
      supervisorStormLocalDir = MesosCommon.DEFAULT_SUPERVISOR_STORM_LOCAL_DIR;
    }
    return supervisorStormLocalDir;
  }
  
  public Map<String, List<TaskInfo>> getTasksToLaunch(Topologies topologies,
                                                      Map<String, Collection<WorkerSlot>> slots,
                                                      Map<String, OfferResources> offerResourcesPerNode) {
    ....
    String supervisorStormLocalDir = getStormLocalDirForWorkers();
    extraConfig += String.format(" -c storm.local.dir=%s", supervisorStormLocalDir);
    ...
  }
```